src/streamlink/plugins/zhanqi.py

```diff
@@ -0,0 +1,84 @@
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import http, validate
+from streamlink.plugin.api.utils import parse_json
+from streamlink.stream import (
+    HTTPStream, HLSStream
+)
+
+API_URL = "http://www.zhanqi.tv/api/static/live.roomid/{roomID[id]}.json"
+
+STATUS_ONLINE = 4
+STATUS_OFFLINE = 0
+
+_url_re = re.compile("""
+    http(s)?://(www\.)?zhanqi.tv
+    /(?P<channel>[^/]+)
+""", re.VERBOSE)
+
+_json_re = re.compile(r"window\.oPageConfig\.oRoom\s*=\s*({.+?});")
+
+_roomID_schema = validate.Schema(
+    validate.all(
+        validate.transform(_json_re.search),
+        validate.any(
+            None,
+            validate.all(
+                validate.get(1),
+                validate.transform(parse_json),
+                {
+                    "id": validate.all(
+                        validate.text,
+                        validate.transform(int)
+                    )
+                }
+            )
+        )
+    )
+)
+
+_room_schema = validate.Schema(
+    {
+        "data": validate.any(None, {
+            "status": validate.all(
+                validate.text,
+                validate.transform(int)
+            ),
+            "videoId": validate.text
+        })
+    },
+    validate.get("data")
+)
+
+
+class Zhanqitv(Plugin):
+    @classmethod
+    def can_handle_url(self, url):
+        return _url_re.match(url)
+
+    def _get_streams(self):
+        match = _url_re.match(self.url)
+        channel = match.group("channel")
+
+        roomID = http.get(self.url, schema=_roomID_schema)
+
+        res = http.get(API_URL.format(roomID=roomID))
+        room = http.json(res, schema=_room_schema)
+        if not room:
+            return
+
+        if room["status"] != STATUS_ONLINE:
+            return
```

Comment:
Maybe a `self.logger.info` message to say that the stream is not currently available?